corehq/apps/app_manager/templatetags/xforms_extras.py

```diff
@@ -100,7 +101,7 @@ def _input_trans(template, name, langs=None, allow_blank=True):
                 options['placeholder'] = name[lang] if (allow_blank or name[lang] != '') else placeholder
                 options['lang'] = lang
             break
-    options = {key: re.sub(r"'", "\\'", value) for (key, value) in options.iteritems()}
+    options = {key: '&quot;' + json.dumps(value)[1:-1] + '&quot;' for (key, value) in options.iteritems()}
```

Comment:
If it gets something like a dict of strings, the code will run and the widget will end up displaying something like `'{"a": "2"}'` which is weird-looking but not really right or wrong.